### DISCUSSION TITLE
v2.0.0

### DISCUSSION THREAD

**Discussion by devcshort** [211-Connect/Norse#138] [Announcements]
`v2.0.0` is coming a long great and is almost ready to be merged in to `main`.  `v2.0.0` is a massive change from the last version that we officially deployed (`v1.0.1`). While we did make some minor changes from then to now, we didn't tag them as releases.

Checkout the latest code to get a head start on our [refactor-main](https://github.com/211-Connect/Norse/tree/refactor-main) branch
Here's a list of things that are being changed in `v2.0.0`

- Moving away from `@mantine` for the UI in favor of `tailwindcss`, `shadcn`, and `radix-ui`. We want people to be able to easily make style changes using a tool that is more well known instead of having to go dig deep in the documentation of a massive component library. We also wanted more control over various components that mantine wasn't able to provide.
- Merging the separate API in to Nextjs. Our APIs aren't that complex, and we don't believe that we need to have a separate API application for this use case. We've found that Nextjs should be scalable to fit our needs, and we believe the same thing will be applicable to most people who use this application. This also makes the application much easier to deploy
- Removing all code that is specific to us and our use cases. These remnants of code that are specific to us have proved to be annoying and confusing for users of the open source code base. We've created our own internal processes so that we can continue to have our own separate features without having to muddy up the open source codebase.
- Migrating to Postgres from MongoDB. We've found MongoDB to be a pain to work with for some of our internal processes. We'll be using Postgres from `v2.0.0` onwards.
- Documentation will be getting a facelift. Our documentation has been lagging behind development. As part of `v2.0.0` we will be giving our documentation a face lift so that it includes better and more up to date instructions for installation, development, and deployment
- Github Actions set up to automatically create and publish an official Docker image for `Norse`
- more?

**Comment by Cskyleryoung** [comment]
*drop jaw*

On Mon, May 27, 2024 at 11:25 PM Chris Short ***@***.***>
wrote:

> v2.0.0 is coming a long great and is almost ready to be merged in to main.
> v2.0.0 is a massive change from the last version that we officially
> deployed (v1.0.1). While we did make some minor changes from then to now,
> we didn't tag them as releases.
>
> Here's a list of things that are being changed in v2.0.0
>
>    - Moving away from @mantine for the UI in favor of tailwindcss, shadcn,
>    and radix-ui. We want people to be able to easily make style changes
>    using a tool that is more well known instead of having to go dig deep in
>    the documentation of a massive component library. We also wanted more
>    control over various components that mantine wasn't able to provide.
>    - Merging the separate API in to Nextjs. Our APIs aren't that complex,
>    and we don't believe that we need to have a separate API application for
>    this use case. We've found that Nextjs should be scalable to fit our needs,
>    and we believe the same thing will be applicable to most people who use
>    this application. This also makes the application much easier to deploy
>    - Removing all code that is specific to us and our use cases. These
>    remnants of code that are specific to us have proved to be annoying and
>    confusing for users of the open source code base. We've created our own
>    internal processes so that we can continue to have our own separate
>    features without having to muddy up the open source codebase.
>    - Migrating to Postgres from MongoDB. We've found MongoDB to be a pain
>    to work with for some of our internal processes. We'll be using Postgres
>    from v2.0.0 onwards.
>    - Documentation will be getting a facelift. Our documentation has been
>    lagging behind development. As part of v2.0.0 we will be giving our
>    documentation a face lift so that it includes better and more up to date
>    instructions for installation, development, and deployment
>    - Github Actions set up to automatically create and publish an
>    official Docker image for Norse
>    - more?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/211-Connect/Norse/discussions/138>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAYZRPKX7JDCGANLA42QAIDZEP2MHAVCNFSM6AAAAABIMCFQV6VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWG42DANRVHA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Skyler Young
Partner/Developer
+1 425 418 8101
Schedule a Call <https://savvycal.com/skyleryoung/call>
sitesavvy.com
